src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -145,8 +147,10 @@ public AgentSettings LoadSettings()
                                                 args,
                                                 enforceSupplied);
                 Trace.Info("serverUrl: {0}", serverUrl);
+                Dictionary<string, string> credentialArgs = (args == null) ? new Dictionary<string, string>() : new Dictionary<string, string>(args);
+                credentialArgs[CliArgs.Url] = serverUrl;
 
-                credProv = AcquireCredentials(args, enforceSupplied);
+                credProv = AcquireCredentials(credentialArgs, enforceSupplied);
```

Comment:
I need to pass ServerUrl to NTLMCredential.ReadCredential and the easiest way was to use "args". One option was to add it to "args", but I thought it may be good to keep "args" readonly for ConfigurationManager.ConfigureAsync. So I made a copy named "credentialArgs" and added "ServerURL".